File: aplicação/frontend/ea-project/src/models/output/GetTeamByIdOutput.js
import BaseOutput from "./BaseOutput";

export default class GetTeamByIdOutput extends BaseOutput {
  constructor(id = 0, name = '', isActive = false, creationDate = new Date(), club = {}, users = [], feedbackMessages = []) {
    super(feedbackMessages);
    this.id = id;
    this.name = name;
    this.isActive = isActive;
    this.creationDate = creationDate;
    this.club = new GetTeamByIdOutput.ClubProperties(club);
    this.users = users.map(item => new GetTeamByIdOutput.UserProperties(item));
  }

  static ClubProperties = class {
    constructor(id = 0, name = '') {
      this.id = id;
      this.name = name;
    }
  };

  static UserProperties = class {
    constructor() {
    }
  };
}

File: app/backend/src/main/java/eaproject/output/GetTeamByIdOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;

public class GetTeamByIdOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private boolean isActive;

    @JsonProperty
    private Timestamp creationDate;

    @JsonProperty
    private ClubProperties club;

    @JsonProperty
    private ArrayList<UserProperties> users;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(boolean isActive) {
        this.isActive = isActive;
    }

    public Timestamp getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }

    public ClubProperties getClub() {
        return club;
    }

    public void setClub(ClubProperties club) {
        this.club = club;
    }

    public ArrayList<UserProperties> getUsers() {
        return users;
    }

    public void setUsers(ArrayList<UserProperties> users) {
        this.users = users;
    }

    public static class ClubProperties implements Serializable {

        @JsonProperty
        private int id;

        @JsonProperty
        private String name;

        public int getId() {
            return id;
        }

        public void setId(int id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    public static class UserProperties implements Serializable {

        @JsonProperty
        private int id;

        @JsonProperty
        private String name;

        public int getId() {
            return id;
        }

        public void setId(int id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }
}
